packages/reactivity/src/effect.ts

```diff
@@ -306,14 +306,19 @@ export function triggerEffects(
 ) {
   pauseScheduling()
   for (const effect of dep.keys()) {
+    // dep.get(effect) is very expensive, we need to calculate it lazily and reuse the result
+    let tracking: boolean | undefined
+
     if (!dep.computed && effect.computed) {
-      if (dep.get(effect) === effect._trackId && effect._runnings > 0) {
+      if (
+        (tracking ??= dep.get(effect) === effect._trackId) &&
+        effect._runnings > 0
```

Comment:
Would it perhaps be better to check `_runnings` first, try to avoid calling `dep.get(effect)` at all?

Reply:
Thanks for pointing that out! I've made the changes.